gcloud/pubsub/subscription.py

```diff
@@ -67,12 +75,16 @@ def from_api_repr(cls, resource, client, topics=None):
         if topics is None:
             topics = {}
         topic_path = resource['topic']
-        topic = topics.get(topic_path)
-        if topic is None:
-            # NOTE: This duplicates behavior from Topic.from_api_repr to avoid
-            #       an import cycle.
-            topic_name = topic_name_from_path(topic_path, client.project)
-            topic = topics[topic_path] = client.topic(topic_name)
+        if topic_path == cls._DELETED_TOPIC_PATH:
+            # Use a name which cannot conflict ('#' not allowed).
+            topic = client.topic(None)
```

Comment:
It just doesn't seem like a good thing to give to a user. They have no reason to think the `Topic` instance is brain-damaged. It'd be nice to have a clearer "don't use this" signal.

The topic / subscription relationship isn't the issue here, right? It's just that the `Subscription` needs to get access to the client (and the project via the client) in **some way**.